packages/proxy/lib/http/util/regex-rewriter.ts

```diff
@@ -1,23 +1,48 @@
+import type { SecurityOpts } from './rewriter'
+
 const pumpify = require('pumpify')
 const { replaceStream } = require('./replace_stream')
 const utf8Stream = require('utf8-stream')
+const integrityStrippedAttributeTag = 'cypress-stripped-integrity'
 
 const topOrParentEqualityBeforeRe = /((?:\bwindow\b|\bself\b)(?:\.|\[['"](?:top|self)['"]\])?\s*[!=]==?\s*(?:(?:window|self)(?:\.|\[['"]))?)(top|parent)(?![\w])/g
 const topOrParentEqualityAfterRe = /(top|parent)((?:["']\])?\s*[!=]==?\s*(?:\bwindow\b|\bself\b))/g
 const topOrParentLocationOrFramesRe = /([^\da-zA-Z\(\)])?(\btop\b|\bparent\b)([.])(\blocation\b|\bframes\b)/g
+
 const jiraTopWindowGetterRe = /(!function\s*\((\w{1})\)\s*{\s*return\s*\w{1}\s*(?:={2,})\s*\w{1}\.parent)(\s*}\(\w{1}\))/g
 const jiraTopWindowGetterUnMinifiedRe = /(function\s*\w{1,}\s*\((\w{1})\)\s*{\s*return\s*\w{1}\s*(?:={2,})\s*\w{1}\.parent)(\s*;\s*})/g
 
-export function strip (html: string) {
-  return html
+const topIsSelfEqualityRe = /(?<=[a-zA-z]\.self==[a-zA-z]\.)top|(?<=[a-zA-z]\.self===[a-zA-z]\.)top|top(?===[a-zA-z]\.self)|top(?====[a-zA-z]\.self)|^top$/g
+const formTopTarget = /target="_top"/g
+// since regex lookbehinds must be a fixed quantifier, we cannot match on script/link tags and need to look ahead to known values
+// this means we might be striping link integrity when we don't need to, especially link tags
+const integrityTagLookAheadRe = /(?<=\s)integrity(?==(?:\"|\')sha(?:256|384|512)-.*?(?:\"|\'))|^integrity$/g
+const dynamicIntegritySetAttributeRe = /(?<=[a-zA-z]\.setAttribute\((?:\"|\'))integrity(?=\"|\')|^integrity$/g
```

Comment:
@AtofStryker has the performance of these regexes been evaluated against large JS/HTML bodies?

Reply:
It has not, which is my concern with using lookback/lookahead. Going to exercise a few tests and see how we do. Might be a good idea to add a some type of test to verify completion in a certain amount of time for a large script/html body?

Reply:
Yeah, you could extend this test: https://github.com/cypress-io/cypress/blob/develop/packages/proxy/test/unit/http/util/regex-rewriter.spec.ts#L243

Reply:
@flotwig I added tests with the flag enabled in. Looks like the performance slowdown is a little less and double for large files 30633047af. It's not terrible but also not great?
![Screen Shot 2022-06-30 at 3 22 53 PM](https://user-images.githubusercontent.com/3980464/176761106-0dd6cde7-2aba-4795-a433-b66ad4321df1.png)

Reply:
There are unit tests that test the regexp performance - because it is very critical to the proxy layer. I'd have to go dig them up - and its possible they aren't extremely sensitive and could "pass" even when performance is minorly degraded. What you need to do is run the unit tests locally and compare the performance before/after modifying any of those regexp's. 

Update: looks like based on your image you found the tests and are factoring in the performance correctly.